File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/SoilTemp/src/java2/SoilTemperature/SoiltemperatureState.java

package SoilTemperature.src.java2.SoilTemperature;

public class SoiltemperatureState
{
    private double SoilSurfaceTemperature;
    private double SnowWaterContent;
    private Double [] SoilTempArray;
    private int AgeOfSnow;
    private double Albedo;
    private Double[] pSoilLayerDepth;
    
    public SoiltemperatureState() { }
    
    public SoiltemperatureState(SoiltemperatureState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.SoilSurfaceTemperature = toCopy.getSoilSurfaceTemperature();
            this.SnowWaterContent = toCopy.getSnowWaterContent();
            this.Albedo = toCopy.getAlbedo();
            this.AgeOfSnow = toCopy.getAgeOfSnow();
            SoilTempArray = new Double[toCopy.getSoilTempArray().length];
            for (int i = 0; i < toCopy.getSoilTempArray().length; i++)
            {
                SoilTempArray[i] = toCopy.getSoilTempArray()[i];
            }

            pSoilLayerDepth = new Double[toCopy.getpSoilLayerDepth().length];
            for (int i = 0; i < toCopy.getpSoilLayerDepth().length; i++)
            {
                pSoilLayerDepth[i] = toCopy.getpSoilLayerDepth()[i];
            }
        }
    }
    public double getSoilSurfaceTemperature()
    { return SoilSurfaceTemperature; }

    public void setSoilSurfaceTemperature(double _SoilSurfaceTemperature)
    { this.SoilSurfaceTemperature= _SoilSurfaceTemperature; } 
    
    public double getSnowWaterContent()
    { return SnowWaterContent; }

    public void setSnowWaterContent(double _SnowWaterContent)
    { this.SnowWaterContent= _SnowWaterContent; }
    
    
    public double getAlbedo()
    { return Albedo; }

    public void setAlbedo(double _Albedo)
    { this.Albedo= _Albedo; } 
    
    public int getAgeOfSnow()
    { return AgeOfSnow; }

    public void setAgeOfSnow(int _AgeOfSnow)
    { this.AgeOfSnow= _AgeOfSnow; }
    
    

    public Double [] getSoilTempArray()
    { return SoilTempArray; }

    public void setSoilTempArray(Double [] _SoilTempArray)
    { this.SoilTempArray= _SoilTempArray; } 
    

    
    public Double [] getpSoilLayerDepth()
    { return pSoilLayerDepth; }

    public void setpSoilLayerDepth(Double [] _pSoilLayerDepth)
    { this.pSoilLayerDepth= _pSoilLayerDepth; } 
    
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/Simplace_SoilTemperature/src/sirius/Simplace_SoilTemperature/SoilTemperatureState.cs

using System;
using System.Collections.Generic;
using CRA.ModelLayer.Core;
using System.Reflection;
using CRA.ModelLayer.ParametersManagement;   

namespace SiriusQualitySoilTemperature.DomainClass
{
    public class SoilTemperatureState : ICloneable, IDomainClass
    {
        private double _Albedo;
        private double _SnowWaterContent;
        private double _SoilSurfaceTemperature;
        private int _AgeOfSnow;
        private double[] _pSoilLayerDepth;
        private double[] _SoilTempArray;
        private ParametersIO _parametersIO;

        public SoilTemperatureState()
        {
            _parametersIO = new ParametersIO(this);
        }

        public SoilTemperatureState(SoilTemperatureState toCopy, bool copyAll) // copy constructor 
        {
            if (copyAll)
            {
                _Albedo = toCopy._Albedo;
                _SnowWaterContent = toCopy._SnowWaterContent;
                _SoilSurfaceTemperature = toCopy._SoilSurfaceTemperature;
                _AgeOfSnow = toCopy._AgeOfSnow;
                pSoilLayerDepth = new double[toCopy._pSoilLayerDepth.Length];
            for (int i = 0; i < toCopy._pSoilLayerDepth.Length; i++)
            { _pSoilLayerDepth[i] = toCopy._pSoilLayerDepth[i]; }
    
                SoilTempArray = new double[toCopy._SoilTempArray.Length];
            for (int i = 0; i < toCopy._SoilTempArray.Length; i++)
            { _SoilTempArray[i] = toCopy._SoilTempArray[i]; }
    
            }
        }

        public double Albedo
        {
            get { return this._Albedo; }
            set { this._Albedo= value; } 
        }
        public double SnowWaterContent
        {
            get { return this._SnowWaterContent; }
            set { this._SnowWaterContent= value; } 
        }
        public double SoilSurfaceTemperature
        {
            get { return this._SoilSurfaceTemperature; }
            set { this._SoilSurfaceTemperature= value; } 
        }
        public int AgeOfSnow
        {
            get { return this._AgeOfSnow; }
            set { this._AgeOfSnow= value; } 
        }
        public double[] pSoilLayerDepth
        {
            get { return this._pSoilLayerDepth; }
            set { this._pSoilLayerDepth= value; } 
        }
        public double[] SoilTempArray
        {
            get { return this._SoilTempArray; }
            set { this._SoilTempArray= value; } 
        }

        public string Description
        {
            get { return "SoilTemperatureState of the component";}
        }

        public string URL
        {
            get { return "http://" ;}
        }

        public virtual IDictionary<string, PropertyInfo> PropertiesDescription
        {
            get { return _parametersIO.GetCachedProperties(typeof(IDomainClass));}
        }

        public virtual Boolean ClearValues()
        {
             _Albedo = default(double);
             _SnowWaterContent = default(double);
             _SoilSurfaceTemperature = default(double);
             _AgeOfSnow = default(int);
             _pSoilLayerDepth = new double[];
             _SoilTempArray = new double[];
            return true;
        }

        public virtual Object Clone()
        {
            IDomainClass myclass = (IDomainClass) this.MemberwiseClone();
            _parametersIO.PopulateClonedCopy(myclass);
            return myclass;
        }
    }
}